util/log.py

```diff
@@ -24,6 +24,13 @@ def error(*args, **kwargs):
     sys.exit(2)
 
 
+# The line_num is expected to be zero-indexed
+def parsing_error_preamble(path, line_num, line):
+    write("")
+    write(f"error reading {path}, line {line_num + 1}:", status="error")
```

Comment:
This could this start with `\n` to remove the need to `write("")`?

Reply:
I don't mind this - it's a bit more explicit that we're writing an empty line